usr/src/uts/common/brand/lx/procfs/lx_prvnops.c

```diff
@@ -1718,6 +1726,121 @@ lxpr_write_pid_tid_comm(lxpr_node_t *lxpnp, struct uio *uio, struct cred *cr,
 	return (0);
 }
 
+/*
+ * static int
+ * lxpr_read_pid_coredump_filter(lxpr_node_t *lxpnp, lxpr_uiobuf_t *uiobuf)
+ *
+ * Reads /proc/<pid>/coredump_filter mask
+ *
+ * Arguments:
+ *      *lxpnp  lxprocfs node that received a read request.
+ *      *uiobuf struct holding output data from read request.
+ */
+static void
+lxpr_read_pid_coredump_filter(lxpr_node_t *lxpnp, lxpr_uiobuf_t *uiobuf)
+{
+	uint32_t filter = LXPR_COREDUMP_FILTER_DEFAULT;
+	lx_proc_data_t *pd;
+	proc_t *p;
+
+	ASSERT3U(lxpnp->lxpr_type, ==, LXPR_PID_COREDUMP_FILTER);
+
+	if ((p = lxpr_lock(lxpnp, NO_ZOMB)) == NULL) {
+		lxpr_uiobuf_seterr(uiobuf, EINVAL);
+		return;
+	}
+
+	ASSERT(MUTEX_HELD(&p->p_lock));
+	if ((pd = ptolxproc(p)) != NULL) {
+		filter = pd->l_coredump_filter;
+	}
+	lxpr_unlock(p);
+
+	lxpr_uiobuf_printf(uiobuf, "%08x\n", filter);
+}
+
+/*
+ * static int
+ * lxpr_write_pid_coredump_filter(lxpr_node_t *lxpnp, uio_t *uiop,
+ *     cred_t *cr, caller_context_t *ct)
+ *
+ * Writes /proc/<pid>/coredump_filter mask
+ *
+ * Arguments:
+ *      *lxpnp  lxprocfs node that received a write request.
+ *      *uio    uio struct holding input data from write request.
+ *      *cr     credential data.
+ *      *ct     who is invoking the operation.
+ *
+ * Returns:
+ *     (0) on success
+ *     (EINVAL | ENXIO ) On failure
+ *
+ * Notes:
+ *  According to Linux CORE(5) man page and Linux kernel documentation:
+ *  https://github.com/torvalds/linux/blob/\
+ *  ba65a4e7120a616d9c592750d9147f6dcafedffa/Documentation/\
+ *  filesystems/proc.rst#L41 the following values are supported:
+ *
+ *  bit 0  Dump anonymous private mappings.
+ *  bit 1  Dump anonymous shared mappings.
+ *  bit 2  Dump file-backed private mappings.
+ *  bit 3  Dump file-backed shared mappings.
+ *  bit 4 (since Linux 2.6.24) Dump ELF headers.
+ *  bit 5 (since Linux 2.6.28) Dump private huge pages.
+ *  bit 6 (since Linux 2.6.28) Dump shared huge pages.
+ *  bit 7 (since Linux 4.4) Dump private DAX pages.
+ *  bit 8 (since Linux 4.4) Dump shared DAX pages.
+ *
+ * We are going to validate the masks, but this has no effect on
+ * the running program, this is just to appease software that rely on
+ * writing/reading to/from this procfs node.
+ */
+static int
+lxpr_write_pid_coredump_filter(lxpr_node_t *lxpnp, uio_t *uiop, cred_t *cr,
+	caller_context_t *ct)
+{
+	lx_proc_data_t *pd;
+	uint32_t filter;
+	char buf[32];
+	int err = 0;
+	size_t olen;
+	proc_t *p;
+
+	ASSERT3U(lxpnp->lxpr_type, ==, LXPR_PID_COREDUMP_FILTER);
+
+	if (uiop->uio_offset != 0)
+		return (EINVAL);
+
+	if (uiop->uio_resid == 0)
+		return (0);
+
+	olen = uiop->uio_resid;
+	if (olen > sizeof (buf) - 1)
+		return (EINVAL);
+
+	bzero(buf, sizeof (buf));
+	if ((err = uiomove(buf, olen, UIO_WRITE, uiop)) != 0)
+		return (err);
+
+	if (ddi_strtoul(buf, NULL, 0, (ulong_t *)&filter) != 0) {
```

Comment:
Yeah, @tsoome is right.  Sorry for missing this.

Suggestion:

- `filter` should be ulong_t.
- Later below you can crunch it down to 32 bits